paddleseg/models/enet.py

```diff
@@ -0,0 +1,631 @@
+        x = self.regular4_1(x)
+        x = self.regular4_2(x)
+
+        # Stage 5 - Decoder
+        x = self.upsample5_0(x, max_indices1_0, output_size=stage1_input_size)
+        x = self.regular5_1(x)
+        x = self.transposed_conv(x, output_size=input_size[2:])
+        return [x]
+
+    def init_weight(self):
+        if self.pretrained is not None:
+            utils.load_pretrained_model(self, self.pretrained)
+
+
+class InitialBlock(nn.Layer):
+    """
+    The initial block is composed of two branches:
+    1. a main branch which performs a regular convolution with stride 2;
+    2. an extension branch which performs max-pooling.
+    Doing both operations in parallel and concatenating their results
+    allows for efficient downsampling and expansion. The main branch
+    outputs 13 feature maps while the extension branch outputs 3, for a
+    total of 16 feature maps after concatenation.
+    Args:
+        in_channels (int): the number of input channels.
+        out_channels (int): the number output channels.
+        kernel_size (int, optional): the kernel size of the filters used in
+            the convolution layer. Default: 3.
+        padding (int, optional): zero-padding added to both sides of the
+            input. Default: 0.
+        bias (bool, optional): Adds a learnable bias to the output if
+            ``True``. Default: False.
+        relu (bool, optional): When ``True`` ReLU is used as the activation
+            function; otherwise, PReLU is used. Default: True.
+    """
+    def __init__(self, in_channels, out_channels, bias=False, relu=True):
+        super(InitialBlock, self).__init__()
+
+        if relu:
+            activation = nn.ReLU
+        else:
+            activation = nn.PReLU
+
+        self.main_branch = nn.Conv2D(in_channels,
+                                     out_channels - 3,
+                                     kernel_size=3,
+                                     stride=2,
+                                     padding=1,
+                                     bias_attr=bias)
+
+        self.ext_branch = nn.MaxPool2D(3, stride=2, padding=1)
+
+        self.batch_norm = layers.SyncBatchNorm(out_channels)
+
+        self.out_activation = activation()
+
+    def forward(self, x):
+        main = self.main_branch(x)
+        ext = self.ext_branch(x)
+
+        out = paddle.concat((main, ext), 1)
+
+        out = self.batch_norm(out)
+
+        return self.out_activation(out)
+
+
+class RegularBottleneck(nn.Layer):
+    """
+    Regular bottlenecks are the main building block of ENet.
+    Main branch:
+    1. Shortcut connection.
+    Extension branch:
+    1. 1x1 convolution which decreases the number of channels by
+        ``internal_ratio``, also called a projection;
+    2. regular, dilated or asymmetric convolution;
+    3. 1x1 convolution which increases the number of channels back to
+        ``channels``, also called an expansion;
+    4. dropout as a regularizer.
+    Args:
```

Comment:
Plz add appropriate blank lines to the note, here and elsewhere.

Reply:
Done.